src/decorators/delay_node.cpp

```diff
@@ -37,19 +36,45 @@ NodeStatus DelayNode::tick()
 
     if (!delay_started_)
     {
+        delay_complete = false;
         delay_started_ = true;
         setStatus(NodeStatus::RUNNING);
         if (msec_ > 0)
         {
-            std::lock_guard<std::timed_mutex> l(delay_mutex);
-            std::thread t(&DelayNode::delay, this);
-            t.join();
+            timer_id_ = timer_.add(std::chrono::milliseconds(msec_), [this](bool aborted) {
```

Comment:
maybe add an extra line like here: https://github.com/BehaviorTree/BehaviorTree.CPP/blob/561f4fa86b45c64c9bb2fd3273f81363e6fd222f/src/decorators/timeout_node.cpp#L58
This way it gets more clear that an inline function is declared here. At least for me.. :)

Reply:
Done